core/src/main/java/com/redhat/lightblue/client/request/data/DataDeleteRequest.java

```diff
@@ -31,6 +31,7 @@ public JsonNode getBodyJson() {
         if (queryExpression != null) {
             node.set("query", queryExpression.toJson());
         }
+        appendExecutionToJson(node);
```

Comment:
This is problematic. AbstractLightblueDataRequest does not know if ArrayNode or ObjectNode is going to be used. This decision is made in child implementation, so I can't populate json body in AbstractLightblueDataRequest.